src/chime_dash/app/utils/templates.py

```diff
@@ -83,16 +83,20 @@ def df_to_html_table(
     dataframe: DataFrame,
     data_only: bool = False,
     n_mod: Optional[int] = None,
-    format: Optional[Dict[Any, str]] = None,
+    formats: Optional[Dict[Any, str]] = None,
 ) -> Table:
     """Converts pandas data frame to html table
     """
-    format = format or {}
+    formats = formats or {}
 
     def cast_type(val):
```

Comment:
Perhaps, cast the whole column as once, instead of casting each column per row: `tmp.date.dt.strffime(DATE_FORMAT)`

If the dtype for the column is correct, map the dtypes to a column casting function. If the dtype is inconsistent (I should fix the model dataframes so they are!), pass in the mappings of column names to the column casting functions to df_to_html_table.

Reply:
I was split between having the more optimal column cast you suggest, assuming that the df is always clean, or having individual type casts. The reason which why i settled for the later where that I encountered a scenario where I wanted to cast a float column to ints (which where actually ints but had NaNs). This was the safest approach I could think of for a general scenario. If performance is relevant, we should probably adjust this.